src/api/apiMethods.js

```diff
@@ -0,0 +1,324 @@
+          columnGroup.props.lastChildIdx = lastChildIndex;
+          columnGroup.props.index = columnGroupIndex;
+
+          columnGroup.props.width =
+            colOffsetIntervalTree.sumTo(columnGroup.props.lastChildIdx) -
+            colOffsetIntervalTree.sumUntil(columnGroup.props.firstChildIdx);
+        }
+
+        return columnGroup;
+      };
+
+      const getColumn = (columnIndex, cellGroupType = 'scrollable') => {
+        const column = _getColumn(columnIndex, cellGroupType);
+        const offset = getColumnOffset(columnIndex, cellGroupType);
+        return _getMinimalColumn(column, offset);
+      };
+
+      const getColumnCount = (cellGroupType = 'scrollable') => {
+        if (cellGroupType === 'fixed') {
+          return columnSettings.fixedColumnsCount;
+        } else if (cellGroupType === 'fixedRight') {
+          return columnSettings.fixedRightColumnsCount;
+        } else {
+          return columnSettings.scrollableColumnsCount;
+        }
+      };
+
+      const getColumnGroupCount = (cellGroupType = 'scrollable') => {
+        if (cellGroupType === 'fixed') {
+          return fixedColumnGroups.length;
+        } else if (cellGroupType === 'fixedRight') {
+          return fixedRightColumnGroups.length;
+        } else {
+          // the parent column group of the last scrollable column will also be the last scrollable column group
+          const lastColumn = _getScrollableColumn(
+            columnSettings.scrollableColumnsCount - 1
+          );
+          return lastColumn.props.columnGroupIndex + 1;
+        }
+      };
+
+      const _getScrollableColumnGroupOffset = (columnGroup) => {
+        // offset of column group is same as offset of its first child column
+        return colOffsetIntervalTree.sumUntil(columnGroup.props.firstChildIdx);
+      };
+
+      const getColumnGroup = (
+        columnGroupIndex,
+        cellGroupType = 'scrollable'
+      ) => {
+        let columnGroup;
+        let offset = 0;
+
+        if (cellGroupType === 'fixed') {
+          columnGroup = fixedColumnGroups[columnGroupIndex];
+          offset = fixedColumnGroupOffsets[columnGroupIndex];
+        } else if (cellGroupType === 'fixedRight') {
+          columnGroup = fixedRightColumnGroups[columnGroupIndex];
+          offset = fixedRightColumnGroupOffsets[columnGroupIndex];
+        } else {
+          columnGroup = _getScrollableColumnGroup(columnGroupIndex);
+          offset = _getScrollableColumnGroupOffset(columnGroup);
+        }
+
+        return _getMinimalColumnGroup(columnGroup, offset);
+      };
+
+      const getColumnGroupByChild = (
+        columnIndex,
+        cellGroupType = 'scrollable'
+      ) => {
+        const column = _getColumn(columnIndex, cellGroupType);
+        const columnGroupIndex = column.props.columnGroupIndex;
+        return getColumnGroup(columnGroupIndex, cellGroupType);
+      };
+
+      const _getElementAtOffset = (container, offset) => {
```

Comment:
We don't maintain `PrefixIntervalTree`s for fixed columns/column groups. So we have to iteratively walk through the collection.
I strongly believe this is completely fine though, because no-one is going to have too many fixed columns without locking up the table because there's won't be any free space available.